packages/suite-base/src/components/SettingsTreeEditor/NodeEditor.test.tsx

```diff
@@ -0,0 +1,101 @@
+/** @jest-environment jsdom */
+
+// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
+// SPDX-License-Identifier: MPL-2.0
+
+import { act, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import "@testing-library/jest-dom";
+
+import { Immutable, SettingsTreeAction, SettingsTreeNode } from "@lichtblick/suite";
+import { NodeEditor } from "@lichtblick/suite-base/components/SettingsTreeEditor/NodeEditor";
+import {
+  FieldEditorProps,
+  SelectVisibilityFilterValue,
+} from "@lichtblick/suite-base/components/SettingsTreeEditor/types";
+import BasicBuilder from "@lichtblick/suite-base/testing/builders/BasicBuilder";
+
+let capturedActionHandler: (action: SettingsTreeAction) => void;
+
+jest.mock("@lichtblick/suite-base/components/SettingsTreeEditor/FieldEditor", () => ({
+  FieldEditor: (props: FieldEditorProps) => {
+    capturedActionHandler = props.actionHandler;
+    return <div />; // Simple mock because UI does not matter here
+  },
+}));
+
+const changeVisibilityFilter = (visibility: SelectVisibilityFilterValue) => {
+  capturedActionHandler({
+    action: "update",
+    payload: { input: "select", value: visibility, path: ["topics", "visibilityFilter"] },
+  });
+};
+
+describe("NodeEditor childNodes filtering", () => {
+  const nodes: [string, string, string] = BasicBuilder.strings({ count: 3 }) as [
+    string,
+    string,
+    string,
+  ];
+
+  const tree: Immutable<SettingsTreeNode> = {
+    enableVisibilityFilter: true,
+    children: {
+      [nodes[0]]: { visible: true, label: nodes[0] },
+      [nodes[1]]: { visible: false, label: nodes[1] },
+      [nodes[2]]: { label: nodes[2] }, // undefined visibility is always shown
+    },
+  };
```

Comment:
or just use `!` in `nodes[x]!`

```suggestion
  const nodes = BasicBuilder.strings();

  const tree: Immutable<SettingsTreeNode> = {
    enableVisibilityFilter: true,
    children: {
      [nodes[0]!]: { visible: true, label: nodes[0] },
      [nodes[1]!]: { visible: false, label: nodes[1] },
      [nodes[2]!]: { label: nodes[2] }, // undefined visibility is always shown
    },
  };
```

Reply:
option 2
I prefer this one

Reply:
lets go with option 1, without forcing the `!` sign